src/dfcx_scrapi/builders/intent.py

```diff
@@ -14,19 +14,467 @@
+
+        return self.proto_obj
+
+
+    def add_training_phrase(
+        self,
+        phrase: Union[str, List[str]],
+        annotations: List[str] = None,
+        repeat_count: int = 1
+    ) -> Intent:
+        """Add a training phrase to proto_obj.
+
+        Args:
+          phrase (string or list of strings):
+            The training phrase as a string without annotations or
+            a list of strings that represents a single training phrase.
+          annotations (list of strings):
+            A list of strings that represents
+              parameter_id of each part in phrase.
+            Length of annotations list should be less than or equal to
+              length of phrase list.
+            If the length is less than length of phrase list it propagate
+              the rest of the annotations automatically with no annotation.
+          repeat_count (int):
+            Indicates how many times this example was added to the intent.
+
+        Example 1: phrase = "I want to check my balance"
+        Example 2:
+          phrase = [
+              'one way', ' ticket leaving ', 'January 1',
+              ' to ', 'LAX', ' from ', 'CDG'
+          ]
+          annotations = [
+              'flight_type', '', 'departure_date',
+              '', 'arrival_city', '', 'departure_city'
+          ]
+        Example 3:
+          phrase = ["I'd like to buy a ", 'one way', ' ticket']
+          annotations = ['', 'flight_type']
+
+        Returns:
+          An Intent object stored in proto_obj
+        """
+
+        self._check_intent_exist()
+
+        # Add simple training phrase
+        if isinstance(phrase, str):
+            # Create the training phrase obj and add it to the others
+            tp = Intent.TrainingPhrase(
+                parts=[Intent.TrainingPhrase.Part(text=phrase)],
+                repeat_count=repeat_count
+            )
+            self.proto_obj.training_phrases.append(tp)
+
+            return self.proto_obj
+
+        # Add annotated training phrase
+        # Type / Error checking
+        if not (isinstance(phrase, list) and isinstance(annotations, list)):
+            raise ValueError(
+                "Both phrase and annotations should be lists."
+            )
+        if not (
+            all((isinstance(p, str) for p in phrase)) and
+            all((isinstance(a, str) for a in annotations))
+        ):
+            raise ValueError(
+                "Only strings allowed in phrase or annotations list."
+            )
+        if len(annotations) > len(phrase):
+            raise IndexError(
+                "Length of annotations list is more than phrase list!"
+            )
+
+        # Propagate the annotations list if needed
+        if len(annotations) < len(phrase):
+            annotations.extend([""] * (len(phrase) - len(annotations)))
+        # Creating parts for the training phrase
+        parts_list = []
+        for text, parameter_id in zip(phrase, annotations):
```

Comment:
@MRyderOC When using `training phrase` as a List and `annotations` together, I encountered some non-intuitive behavior. 

Essentially, my assumption was that I would add each "part" of my Training Phrase as an item in the list
`['my', 'name', 'is', 'Patrick']`

And each of my annotations in another list:  
`['', '', '', 'people']`

However, the end result was the following phrase:  
`mynameisPatrick`

My thought is that we should concat spaces for the user by default so that it makes the list creation / spacing less error prone. If the user forgets to space appropriately, this will "fail silently" in that it will successfully create the Training Phrase, but not with the intended result that the user wanted. At scale, this could be quite bothersome.

WDYT?

Reply:
This is a totally valid point. Despite the fact that this behavior is influenced by the base library's [docs](https://github.com/googleapis/python-dialogflow-cx/blob/181e6e333133b27142b95f8dd81d3b3ec78034da/google/cloud/dialogflowcx_v3beta1/types/intent.py#L127-L129) as you always stated we should remove the hassle from the user perspective.
We can add `include_spaces` argument to the function so it can do both.

Reply:
> We can add include_spaces argument to the function so it can do both.

Great idea! 💡  
Let's add `include_spaces=True` as the default value for that parameter.  Then if a user needs to override it they can flip to `False` and do what they need.